keps/sig-network/0015-20180614-SCTP-support.md

```diff
@@ -297,11 +298,71 @@ We describe in the Kubernetes documentation the mutually exclusive nature of use
 
 ### Risks and Mitigations
 
+## Test Plan
+
+Unlike with UDP and TCP, we can't necessarily just test that SCTP
+connections work, because some machines won't have the SCTP kernel
+module installed (or will have it blocked from being loaded). So we
+want to have some tests that are just "make sure kube-proxy is doing
+what we expect it to" that can run everywhere, and then another set of
+"SCTP connections actually work" tests that are behind a feature tag.
+
+For the standard PR/periodic tests, we'll add:
+
+- Something like `"Pods, Services, and NetworkPolicies can declare
+  SCTP ports"`.
+
+  - If using the `iptables` proxier, this will additionally check that
+    appropriate iptables rules are created.
+
+  - The test also ensures that if `sctp.ko` was not loaded before the
+    test (on the node with the created test pod), then it is still not
+    loaded after the test. (FIXME: that could fail if it ran in
+    parallel with some of the other tests below.)
+
+  - (FIXME: how does e2e work with feature gates? This test and the
+    ones below will all require the `SCTPSupport` feature gate to be
+    enabled.)
+
+Additionally, we'll have some tests behind a tag like
+"`[Feature:SCTPConnectivity]`" that can be run by network plugin
+developers but which will not run as part of any of the standard
+kubernetes test runs:
+
+- `"A pod can connect directly to another pod via SCTP"`
+
+- `"A pod can connect to another pod via SCTP via a Service IP"`
+
+- `"A pod can connect to another pod via SCTP via a Load Balancer"`,
+  which will use `framework.SkipUnlessProviderIs` to skip clouds that
+  don't support SCTP LBs.
+
 ## Graduation Criteria
 
-## Implementation History
+### Alpha -> Beta Graduation
+
+Graduation criteria:
+
+- The e2e tests described in the Test Plan have been written, and the
+  "can declare SCTP ports" test passes. (FIXME: unless feature gate
+  rules make that impossible)
 
-## Drawbacks [optional]
+### Beta -> GA Graduation
 
-## Alternatives [optional]
+Graduation criteria:
+
+- At least 2 external network plugins can pass the
```

Comment:
define "external" ?

Reply:
"not kubenet". external to kubernetes/kubernetes.